File: 104-MaximumDepthOfBinaryTree/104-MaximumDepthOfBinaryTree.py
# Last updated: 08/05/2026, 09:32:18
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        
        def recur(node, leng):
            if node==None:
                return leng
            leng+=1
            r=recur(node.right,leng)
            l=recur(node.left,leng)
            
            print("r:",r," l",l)
            return r if r>l else l

        return recur(root,0)

File: 104-MaximumDepthOfBinaryTree/104-MaximumDepthOfBinaryTree.cpp
// Last updated: 08/05/2026, 09:32:22
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    int dfs( TreeNode* node){
        if(!node){
            return 0;
        }

         int val=max(dfs(node->left), dfs(node->right));

         return val+1;
    }

    int maxDepth(TreeNode* root) {
        return dfs(root);
        
    }
};